assets/app/scripts/services/applicationGenerator.js

```diff
@@ -202,6 +202,13 @@ angular.module("openshiftConsole")
           type: "ImageChange"
         });
       }
+
+      // User can input a URL that contains a ref
+      var urlAndRef = input.buildConfig.sourceUrl.split("#");
```

Comment:
Yeah I hesitated to do that for the same reason.  Although it will properly parse the fragment from git@example.com:dir1/dir2#test it is just treating everything before the '#' as the path.